tests/dynamic_sharing_service/dynamic.scenario.builder.spec.js

```diff
@@ -81,4 +72,56 @@ describe('DynamicScenarioBuilder', () => {
     expect(dynamicScenario).to.be.instanceOf(DynamicScenario);
     expect(JSON.stringify(dynamicScenario)).to.equal(expectedJson);
   });
+
+  it('should not have selfie or pin auth', () => {
+    const dynamicPolicy = new DynamicPolicyBuilder()
+      .withSelfieAuthentication(true)
+      .withSelfieAuthentication(false)
+      .withPinAuthentication(true)
+      .withPinAuthentication(false)
+      .build();
+
+    const authTypes = dynamicPolicy.getWantedAuthTypes();
+    expect(authTypes).to.not.contain(1);
+    expect(authTypes).to.not.contain(2);
+  });
+
+  it('should not have more than one auth type', () => {
+    const dynamicPolicy = new DynamicPolicyBuilder()
+      .withSelfieAuthentication(true)
+      .withSelfieAuthentication()
+      .withSelfieAuthentication(true)
+      .build();
+
+    const authTypesLength = dynamicPolicy.getWantedAuthTypes().length;
+    expect(authTypesLength).to.equal(1);
+  });
+
+  it('should only have two auth types', () => {
+    const dynamicPolicy = new DynamicPolicyBuilder()
+      .withSelfieAuthentication(true)
+      .withPinAuthentication(true)
+      .build();
+
+    const authTypesLength = dynamicPolicy.getWantedAuthTypes().length;
+    expect(authTypesLength).to.equal(2);
+  });
+
+  it('should not have selfie authentication after having it added then removed', () => {
+    const dynamicPolicy = new DynamicPolicyBuilder()
+      .withSelfieAuthentication(true)
+      .withSelfieAuthentication(false)
+      .build();
+
+    expect(dynamicPolicy.wantedAuthTypes).to.not.contain(1);
+  });
+
+  it('should not have pin authentication after having it added then removed', () => {
+    const dynamicPolicy = new DynamicPolicyBuilder()
+      .withPinAuthentication(true)
+      .withPinAuthentication(false)
+      .build();
+
+    expect(dynamicPolicy.wantedAuthTypes).to.not.contain(1);
```

Comment:
Can we use the `PIN_AUTH_TYPE` and `SELFIE_AUTH_TYPE` consts here throughout the tests, instead of referring to their int values? 👍 This will be easier to manage if we change them in the future, and also makes the tests a bit clearer

Reply:
The auth types aren't currently exported to be available to any other file.  We could move these into the class itself as static variables, but checking the Java SDK these are private as well.  I think it would be good to give visibility to the rest of the application for these, but it's a decision that needs to be made.

Reply:
We could move these constants into the constructor for the class therefore making them accessible to other components:

`
this.SELFIE_AUTH_TYPE = 1;
this.PIN_AUTH_TYPE = 2;
`

I tried using a 'get' function but eslint didn't like the fact that it wasn't using 'this' inside of a class method.

Currently the module only exports the class so without changing how the module export works I think this is the best solution (as the constants are directly related to the policy builder)

Reply:
After some more looking at this, I think the constants would probably be better off as static 'get' functions within the DynamicPolicy class (rather than the builder which should focus solely on building an actual DynamicPolicy).  Example below:
`
get SELFIE_AUTH_TYPE() {
  return 1;
}

// Call with the following
DynamicPolicy.SELFIE_AUTH_TYPE
`

Thoughts @echarrod ?

Reply:
Seems ok to me, but I think we can wait for @davidgrayston to get back tomorrow to get his thoughts 👌